src/nvim/ex_getln.c

```diff
@@ -512,8 +512,12 @@ static int command_line_execute(VimState *state, int key)
   CommandLineState *s = (CommandLineState *)state;
   s->c = key;
 
-  if (s->c == K_EVENT) {
-    multiqueue_process_events(main_loop.events);
+  if (s->c == K_EVENT || s->c == K_COMMAND) {
+    if (s->c == K_EVENT) {
+      multiqueue_process_events(main_loop.events);
+    } else {
+      do_cmdline(NULL, getcmdkeycmd, NULL, DOCMD_NOWAIT);
```

Comment:
Why DOCMD_NOWAIT here, but `0` in the `insert_handle_key` case?

Reply:
Because it displays error messages in a way that makes sense in the respective mode.